### DISCUSSION TITLE
How to update .env variables on my production environment / server?

### DISCUSSION THREAD

**Discussion by ralphjsmit** [envault/envault#57] [Q&A]
Hey Envault! A few minutes ago I stumbled across this repository on GitHub. Managing .env files is something that is highly relevant to me, as I'm still trying to find a good way to manage them. 

After reading the docs and watching the video, I understand that (please correct if I'm wrong):

1. I install a separate app, e.g. on a personal subdomain
2. In that app I can update and change .env variables
3. If I want to sync those variables to my local installation, I can use npx

My question is: how do I get the variables in my production environments? Do I need npm installed on my webserver as well?

In short, I'm using deployer.org to deploy my applications to my server via SSH / Git. I now need to find a way to update my .env file with new variables (when deploying), **without deleting my APP_KEY** .

How do I update an .env file on my production (during deploy)? Unfortunately my server doesn't support npm.

Many thanks in advance!

**Comment by danharrin** [comment]
Not sure what integration you're looking for, we currently only support NPX so its difficult to use Envault without it. Is there some sort of .env API for Deployer?

**Reply by ralphjsmit** [reply]
Hey Dan, thanks for coming back to me. No, there's no sort of .env API for deployer. You have to write/make that yourself. 
That's why I also really liked envault, till I discovered that my hosting doesn't support npm. 

It'd be awesome if you could create some sort of functionality to sync the .env without npx.

For now I solved it by just committing an .env.prod and an .env.dev file, and doing a few shell scripts to move/copy/find and replace. (I know that committing .env files is bad practice, but it's solo project, so good for now.)

**Reply by danharrin** [reply]
Hey, Envault has an exposed API you could use, its not documented but you could look at the code in the [CLI repo](https://github.com/envault/cli) and work out how to contact the API and update environment variables using a shell script.

Good luck, and be sure to share if you get a pure-shell script working!

**Reply by ralphjsmit** [reply]
Ah, that is really helpful. 

I had a quick look. In the below code, is `vault.envault.dev` the server, `84632` the environment and `iCNaGGLou6v0mRas` the token?
```
npx envault vault.envault.dev 84632 iCNaGGLou6v0mRas
```

**Reply by danharrin** [reply]
Exactly!

**Reply by ralphjsmit** [reply]
Perhaps it would be good to first implement #23 (multiple environments) first. Without support for multiple environments, a shell script would not be very useful yet. And when that's supported I can immediately take that into account.

**Reply by danharrin** [reply]
> Perhaps it would be good to first implement #23 (multiple environments) first. Without support for multiple environments, a shell script would not be very useful yet. And when that's supported I can immediately take that into account.

At the moment, you can just use a different app for each environment. Multi-environment support will come soon, but I have other projects to prioritise first that I'll use to build v3.

**Reply by ralphjsmit** [reply]
Of course, understandable! I absolutely don't want to force anything on you (I'm already happy that you created this!), but any idea when that could be?

**Reply by danharrin** [reply]
I'm super super busy with the new version of [Filament](https://github.com/laravel-filament/filament), which will be finished by the end of this year (I know, it's ages away but it's a massive upgrade and I have a full time job on top haha). However, when this is complete, I'll be able to rebuild Envault on top of Filament which will make the development process really fast. Really sorry if this is disappointing, but I think its for the best! I can of course work on bug fixes and security patches here in the meantime, but no major new features like environments until then unfortunately.

**Reply by ralphjsmit** [reply]
I think that I would do absolutely the same. When I look at what open source things you've produced (besides your regular work), you're already doing a terrific job on all that open source software. Curious for the new version of Filament! Appreciate it that you're still putting in time and energy to get Filament to a new version, rather than focus on some new (perhaps more interesting) sideproject.

**Reply by danharrin** [reply]
Thanks! Encouragement like that means a lot to me. 🙏

**Reply by ralphjsmit** [reply]
👏 What main features will you bring to the next version of Filament? It’s a very ingenuous concept. 

Are you also considering a redesign of the UI? It feels slightly outdated and not really anymore in line with the current Tailwind (UI) design. 

Does Filament accept contributions (in the form of a pull request)? If you (and your collaborators) are open to it, I’ll be able to work on an updated design / layout this fall.

**Reply by danharrin** [reply]
All features we plan to support are listed in [this thread](https://github.com/laravel-filament/filament/discussions/408).

And yes we do support PRs and I agree the current UI is slightly outdated, but this will be done in v2 by us :) We're going for a more minimalist "Vercel" aesthetic.

**Reply by ralphjsmit** [reply]
Okay that's awesome! Thanks🙏
